### PR TITLE
Add multiple camera support (Xperia 10 II)

### PR DESCRIPTION
**Pull request by llelectronics**
This adds support for multiple cameras like on the Xperia 10 II by adding 3 buttons to switch to tele, wide and ultra wide lenses while still supporting 2 camera systems. 
In video mode during recording switching between cameras is not possible. This is why the buttons disappear as soon as you record. You can switch cameras before recording however.

Currently the selfie cam is not working in photo mode on the Xperia 10 II. 
In video mode however it is working fine. 
When switching to the selfie cam it saves the last rear lense you used and switches to it when pressing the switch button again.

### CODE REVIEW

File: qml/pages/Settings.qml [hunk 1]
```diff
@@ -11,7 +11,7 @@ Item {
     ConfigurationGroup {
         id: globalSettings
         path: "/uk/co/piggz/harbour-advanced-camera"
-        property int cameraCount: 2
+        property int cameraCount: QtMultimedia.availableCameras.length
         property string cameraId: "primary"
```
Comment (mlehtima) [inline qml/pages/Settings.qml:15]: Should this be also changed to match the new cameraId format?

Comment (llelectronics) [reply qml/pages/Settings.qml:15]: True that makes sense

**Comment by piggz** [comment]
Thanks!  Could you add a check the the spec for the appropriate QtMultimedia version, this would prevent the update going onto older releases of sfos.

**Comment by spag** [comment]
On startup, camera shows image from the previously selected camera for maybe 1/2 second then screen goes black.

Version: harbour-advanced-camera-0.8.0-1.3.aarch64.rpm

```
[D] unknown:0 - Supported  "image"  resolutions: QMap(("1024x768", QSize(1024, 768))("1200x1200", QSize(1200, 1200))("1280x400", QSize(1280, 400))("1280x480", QSize(1280, 480))("1280x768", QSize(1280, 768))("1280x960", QSize(1280, 960))("1440x1080", QSize(1440, 1080))("1600x1200", QSize(1600, 1200))("176x144", QSize(176, 144))("1920x1080", QSize(1920, 1080))("2048x1536", QSize(2048, 1536))("240x320", QSize(240, 320))("2560x800", QSize(2560, 800))("2592x1944", QSize(2592, 1944))("2688x1512", QSize(2688, 1512))("3200x2400", QSize(3200, 2400))("320x240", QSize(320, 240))("3264x2448", QSize(3264, 2448))("352x288", QSize(352, 288))("480x320", QSize(480, 320))("480x360", QSize(480, 360))("480x640", QSize(480, 640))("640x360", QSize(640, 360))("640x640", QSize(640, 640))("720x480", QSize(720, 480))("800x600", QSize(800, 600)))
[D] unknown:0 - Supported  "image"  resolutions: QMap(("1024x768", QSize(1024, 768))("1200x1200", QSize(1200, 1200))("1280x400", QSize(1280, 400))("1280x480", QSize(1280, 480))("1280x768", QSize(1280, 768))("1280x960", QSize(1280, 960))("1440x1080", QSize(1440, 1080))("1600x1200", QSize(1600, 1200))("176x144", QSize(176, 144))("1920x1080", QSize(1920, 1080))("2048x1536", QSize(2048, 1536))("240x320", QSize(240, 320))("2560x800", QSize(2560, 800))("2592x1944", QSize(2592, 1944))("2688x1512", QSize(2688, 1512))("3200x2400", QSize(3200, 2400))("320x240", QSize(320, 240))("3264x2448", QSize(3264, 2448))("352x288", QSize(352, 288))("480x320", QSize(480, 320))("480x360", QSize(480, 360))("480x640", QSize(480, 640))("640x360", QSize(640, 360))("640x640", QSize(640, 640))("720x480", QSize(720, 480))("800x600", QSize(800, 600)))
[D] unknown:0 - Supported  "image"  resolutions: QMap(("1024x768", QSize(1024, 768))("1200x1200", QSize(1200, 1200))("1280x400", QSize(1280, 400))("1280x480", QSize(1280, 480))("1280x768", QSize(1280, 768))("1280x960", QSize(1280, 960))("1440x1080", QSize(1440, 1080))("1600x1200", QSize(1600, 1200))("176x144", QSize(176, 144))("1920x1080", QSize(1920, 1080))("2048x1536", QSize(2048, 1536))("240x320", QSize(240, 320))("2560x800", QSize(2560, 800))("2592x1944", QSize(2592, 1944))("2688x1512", QSize(2688, 1512))("3200x2400", QSize(3200, 2400))("320x240", QSize(320, 240))("3264x2448", QSize(3264, 2448))("352x288", QSize(352, 288))("480x320", QSize(480, 320))("480x360", QSize(480, 360))("480x640", QSize(480, 640))("640x360", QSize(640, 360))("640x640", QSize(640, 640))("720x480", QSize(720, 480))("800x600", QSize(800, 600)))
[D] unknown:0 - "" QQuickGridLayoutBase::rearrange() QSizeF(140, 1060)
[D] onCameraStatusChanged:205 - Camera status: 8
[D] unknown:0 - Supported  "image"  resolutions: QMap(("1024x768", QSize(1024, 768))("1200x1200", QSize(1200, 1200))("1280x400", QSize(1280, 400))("1280x480", QSize(1280, 480))("1280x768", QSize(1280, 768))("1280x960", QSize(1280, 960))("1440x1080", QSize(1440, 1080))("1600x1200", QSize(1600, 1200))("176x144", QSize(176, 144))("1920x1080", QSize(1920, 1080))("2048x1536", QSize(2048, 1536))("240x320", QSize(240, 320))("2560x800", QSize(2560, 800))("2592x1944", QSize(2592, 1944))("2688x1512", QSize(2688, 1512))("3200x2400", QSize(3200, 2400))("320x240", QSize(320, 240))("3264x2448", QSize(3264, 2448))("352x288", QSize(352, 288))("480x320", QSize(480, 320))("480x360", QSize(480, 360))("480x640", QSize(480, 640))("640x360", QSize(640, 360))("640x640", QSize(640, 640))("720x480", QSize(720, 480))("800x600", QSize(800, 600)))
[D] onCameraStatusChanged:205 - Camera status: 7
[D] applySettings:705 - Applying settings in image mode for 2 camera with status 7
[D] onResolutionChanged:150 - Image resolution changed: QSize(3264, 2448)
[D] onResolutionChanged:185 - Video resolution changed: QSize(1920, 1080)
[D] onCameraStatusChanged:205 - Camera status: 1
[D] onCameraStatusChanged:205 - Camera status: 2
[D] onCameraStatusChanged:205 - Camera status: 4

** (harbour-advanced-camera:21600): WARNING **: 09:07:52.789: capsfilter2: transform_caps returned caps which are not a real subset of the filter caps

** (harbour-advanced-camera:21600): WARNING **: 09:07:52.790: capsfilter2: transform_caps returned caps which are not a real subset of the filter caps

** (harbour-advanced-camera:21600): WARNING **: 09:07:52.790: imagebin-capsfilter: transform_caps returned caps which are not a real subset of the filter caps

(harbour-advanced-camera:21600): GStreamer-CRITICAL **: 09:07:52.809: gst_buffer_pool_set_active: assertion 'GST_IS_BUFFER_POOL (pool)' failed

(harbour-advanced-camera:21600): GStreamer-CRITICAL **: 09:07:52.809: gst_buffer_pool_set_active: assertion 'GST_IS_BUFFER_POOL (pool)' failed
[D] onCameraStatusChanged:205 - Camera status: 6
[D] unknown:0 - Supported  "image"  resolutions: QMap(("1024x768", QSize(1024, 768))("1200x1200", QSize(1200, 1200))("1280x400", QSize(1280, 400))("1280x480", QSize(1280, 480))("1280x768", QSize(1280, 768))("1280x960", QSize(1280, 960))("1440x1080", QSize(1440, 1080))("1600x1200", QSize(1600, 1200))("176x144", QSize(176, 144))("1920x1080", QSize(1920, 1080))("2048x1536", QSize(2048, 1536))("240x320", QSize(240, 320))("2560x800", QSize(2560, 800))("2592x1944", QSize(2592, 1944))("2688x1512", QSize(2688, 1512))("3200x2400", QSize(3200, 2400))("320x240", QSize(320, 240))("3264x2448", QSize(3264, 2448))("352x288", QSize(352, 288))("480x320", QSize(480, 320))("480x360", QSize(480, 360))("480x640", QSize(480, 640))("640x360", QSize(640, 360))("640x640", QSize(640, 640))("720x480", QSize(720, 480))("800x600", QSize(800, 600)))
[D] unknown:0 - Supported  "image"  resolutions: QMap(("1024x768", QSize(1024, 768))("1200x1200", QSize(1200, 1200))("1280x400", QSize(1280, 400))("1280x480", QSize(1280, 480))("1280x768", QSize(1280, 768))("1280x960", QSize(1280, 960))("1440x1080", QSize(1440, 1080))("1600x1200", QSize(1600, 1200))("176x144", QSize(176, 144))("1920x1080", QSize(1920, 1080))("2048x1536", QSize(2048, 1536))("240x320", QSize(240, 320))("2560x800", QSize(2560, 800))("2592x1944", QSize(2592, 1944))("2688x1512", QSize(2688, 1512))("3200x2400", QSize(3200, 2400))("320x240", QSize(320, 240))("3264x2448", QSize(3264, 2448))("352x288", QSize(352, 288))("480x320", QSize(480, 320))("480x360", QSize(480, 360))("480x640", QSize(480, 640))("640x360", QSize(640, 360))("640x640", QSize(640, 640))("720x480", QSize(720, 480))("800x600", QSize(800, 600)))

[D] unknown:0 - Supported  "image"  resolutions: QMap(("1024x768", QSize(1024, 768))("1200x1200", QSize(1200, 1200))("1280x400", QSize(1280, 400))("1280x480", QSize(1280, 480))("1280x768", QSize(1280, 768))("1280x960", QSize(1280, 960))("1440x1080", QSize(1440, 1080))("1600x1200", QSize(1600, 1200))("176x144", QSize(176, 144))("1920x1080", QSize(1920, 1080))("2048x1536", QSize(2048, 1536))("240x320", QSize(240, 320))("2560x800", QSize(2560, 800))("2592x1944", QSize(2592, 1944))("2688x1512", QSize(2688, 1512))("3200x2400", QSize(3200, 2400))("320x240", QSize(320, 240))("3264x2448", QSize(3264, 2448))("352x288", QSize(352, 288))("480x320", QSize(480, 320))("480x360", QSize(480, 360))("480x640", QSize(480, 640))("640x360", QSize(640, 360))("640x640", QSize(640, 640))("720x480", QSize(720, 480))("800x600", QSize(800, 600)))
[W] unknown:0 - CameraBin error: "failed to negotiate caps"
[D] onCameraStatusChanged:205 - Camera status: 1
```

**Comment by piggz** [comment]
Try 0.8.1

**Comment by spag** [comment]
0.8.1 fixes this issue for all three back cameras.
Switching from back to front (selfie) camera still makes the camera view go black.
After restarting the app the front camera appears so be working, though.
Switching from front (selfie) camera to either of the back cameras works without the need of a restart.

**Comment by piggz** [comment]
Is this the 10ii?  I think @abranson mentioned this would get fixed before final sfos 4.1

**Comment by spag** [comment]
Yes, Xperia 10 II on Sailfish 4.1.0.23
Switching back to front camera and vice versa works with the stock camera app, though.

**Comment by llelectronics** [comment]
Its an upstream bug for third party apps in Sailfish OS and will be fixed with 4.1.0.24

**Comment by spag** [comment]
I can confirm issue with switching from back to front camera does not occur any more after updating to Sailfish 4.1.0.24.

**Pull request merged**